File: legacy/elementary/src/examples/spinner_example.c
/**
 * Simple Elementary's <b>spinner widget</b> example, illustrating its
 * usage and API.
 *
 * See stdout/stderr for output. Compile with:
 *
 * @verbatim
 * gcc -g `pkg-config --cflags --libs elementary` spinner_example.c -o spinner_example
 * @endverbatim
 */

#include <Elementary.h>
#ifdef HAVE_CONFIG_H
# include "elementary_config.h"
#else
# define __UNUSED__
#endif

static void
_on_done(void *data __UNUSED__,
        Evas_Object *obj __UNUSED__,
        void *event_info __UNUSED__)
{
   elm_exit();
}

static void
_changed_cb(void *data __UNUSED__, Evas_Object *obj, void *event_info __UNUSED__)
{
   printf("Value changed to %0.f\n", elm_spinner_value_get(obj));
}

static void
_delay_changed_cb(void *data __UNUSED__, Evas_Object *obj, void *event_info __UNUSED__)
{
   printf("Value delay changed to %0.f\n", elm_spinner_value_get(obj));
}

int
elm_main(int argc __UNUSED__, char **argv __UNUSED__)
{
   Evas_Object *win, *bg, *bx, *sp;

   win = elm_win_add(NULL, "spinner", ELM_WIN_BASIC);
   elm_win_title_set(win, "Spinner Example");
   evas_object_smart_callback_add(win, "delete,request", _on_done, NULL);

   bg = elm_bg_add(win);
   elm_win_resize_object_add(win, bg);
   evas_object_size_hint_weight_set(bg, EVAS_HINT_EXPAND, EVAS_HINT_EXPAND);
   evas_object_show(bg);

   bx = elm_box_add(win);
   evas_object_size_hint_weight_set(bx, EVAS_HINT_EXPAND, EVAS_HINT_EXPAND);
   elm_win_resize_object_add(win, bx);
   evas_object_show(bx);

   /* default */
   sp = elm_spinner_add(win);
   evas_object_size_hint_align_set(sp, EVAS_HINT_FILL, 0.5);
   evas_object_size_hint_weight_set(sp, EVAS_HINT_EXPAND, EVAS_HINT_EXPAND);
   elm_box_pack_end(bx, sp);
   evas_object_show(sp);

   /* format */
   sp = elm_spinner_add(win);
   elm_spinner_label_format_set(sp, "Percentage %%%1.2f something");
   evas_object_size_hint_align_set(sp, EVAS_HINT_FILL, 0.5);
   evas_object_size_hint_weight_set(sp, EVAS_HINT_EXPAND, EVAS_HINT_EXPAND);
   elm_box_pack_end(bx, sp);
   evas_object_show(sp);

   /* min max, step and wrap */
   sp = elm_spinner_add(win);
   elm_spinner_label_format_set(sp, "%1.1f units");
   elm_spinner_step_set(sp, 1.5);
   elm_spinner_wrap_set(sp, EINA_TRUE);
   elm_spinner_min_max_set(sp, -50.0, 250.0);
   evas_object_size_hint_align_set(sp, EVAS_HINT_FILL, 0.5);
   evas_object_size_hint_weight_set(sp, EVAS_HINT_EXPAND, EVAS_HINT_EXPAND);
   elm_box_pack_end(bx, sp);
   evas_object_show(sp);

   /* vertical */
   sp = elm_spinner_add(win);
   elm_object_style_set(sp, "vertical");
   elm_spinner_interval_set(sp, 0.2);
   evas_object_size_hint_align_set(sp, EVAS_HINT_FILL, 0.5);
   evas_object_size_hint_weight_set(sp, EVAS_HINT_EXPAND, EVAS_HINT_EXPAND);
   elm_box_pack_end(bx, sp);
   evas_object_show(sp);

   /* disabled edition */
   sp = elm_spinner_add(win);
   elm_spinner_editable_set(sp, EINA_FALSE);
   evas_object_size_hint_align_set(sp, EVAS_HINT_FILL, 0.5);
   evas_object_size_hint_weight_set(sp, EVAS_HINT_EXPAND, EVAS_HINT_EXPAND);
   elm_box_pack_end(bx, sp);
   evas_object_show(sp);

   /* special values */
   sp = elm_spinner_add(win);
   elm_spinner_editable_set(sp, EINA_FALSE);
   elm_spinner_min_max_set(sp, 1, 12);
   elm_spinner_special_value_add(sp, 1, "January");
   elm_spinner_special_value_add(sp, 2, "February");
   elm_spinner_special_value_add(sp, 3, "March");
   elm_spinner_special_value_add(sp, 4, "April");
   elm_spinner_special_value_add(sp, 5, "May");
   elm_spinner_special_value_add(sp, 6, "June");
   elm_spinner_special_value_add(sp, 7, "July");
   elm_spinner_special_value_add(sp, 8, "August");
   elm_spinner_special_value_add(sp, 9, "September");
   elm_spinner_special_value_add(sp, 10, "October");
   elm_spinner_special_value_add(sp, 11, "November");
   elm_spinner_special_value_add(sp, 12, "December");
   evas_object_size_hint_align_set(sp, EVAS_HINT_FILL, 0.5);
   evas_object_size_hint_weight_set(sp, EVAS_HINT_EXPAND, EVAS_HINT_EXPAND);
   elm_box_pack_end(bx, sp);
   evas_object_show(sp);

   sp = elm_spinner_add(win);
   evas_object_size_hint_align_set(sp, EVAS_HINT_FILL, 0.5);
   evas_object_size_hint_weight_set(sp, EVAS_HINT_EXPAND, EVAS_HINT_EXPAND);
   elm_box_pack_end(bx, sp);
   evas_object_show(sp);
   evas_object_smart_callback_add(sp, "changed", _changed_cb, NULL);
   evas_object_smart_callback_add(sp, "delay,changed", _delay_changed_cb, NULL);

   evas_object_show(win);

   elm_run();
   return 0;
}
ELM_MAIN()

File: legacy/elementary/src/examples/spinner_example.js

elm = require('elm');

win = new elm.Elm.WinStandard(null);
win.setTitle("Spinner Example");
win.setAutohide(true);

bx = new elm.Elm.Box(win);
bx.setSizeHintWeight(1.0, 1.0);
win.resizeObjectAdd(bx);
bx.setVisible(true);

sp = new elm.Elm.Spinner(win);
sp.setSizeHintWeight(1.0, 1.0);
sp.setSizeHintAlign(-1.0, 0.5);
bx.packEnd(sp);
sp.setVisible(true);

sp2 = new elm.Elm.Spinner(win);
sp2.setLabelFormat("Percentage %%%1.2f something");
sp2.setSizeHintWeight(1.0, 1.0);
sp2.setSizeHintAlign(-1.0, 0.5);
bx.packEnd(sp2);
sp2.setVisible(true);

sp3 = new elm.Elm.Spinner(win);
sp3.setLabelFormat("%1.1f units");
sp3.setStep(1.5);
sp3.setWrap(true);
sp3.setMinMax(-50.0, 250.0);
sp3.setSizeHintWeight(1.0, 1.0);
sp3.setSizeHintAlign(-1.0, 0.5);
bx.packEnd(sp3);
sp3.setVisible(true);

sp4 = new elm.Elm.Spinner(win);
sp4.setStyle("vertical");
sp4.setInterval(0.2);
sp4.setSizeHintWeight(1.0, 1.0);
sp4.setSizeHintAlign(-1.0, 0.5);
bx.packEnd(sp4);
sp4.setVisible(true);

sp5 = new elm.Elm.Spinner(win);
sp5.setEditable(false);
sp5.setSizeHintWeight(1.0, 1.0);
sp5.setSizeHintAlign(-1.0, 0.5);
bx.packEnd(sp5);
sp5.setVisible(true);

sp6 = new elm.Elm.Spinner(win);
sp6.setEditable(false);
sp6.setMinMax(1, 12);
sp6.specialValueAdd(1, "January");
sp6.specialValueAdd(2, "February");
sp6.specialValueAdd(3, "March");
sp6.specialValueAdd(4, "April");
sp6.specialValueAdd(5, "May");
sp6.specialValueAdd(6, "June");
sp6.specialValueAdd(7, "July");
sp6.specialValueAdd(8, "August");
sp6.specialValueAdd(9, "September");
sp6.specialValueAdd(10, "October");
sp6.specialValueAdd(11, "November");
sp6.specialValueAdd(12, "December");
sp6.setSizeHintWeight(1.0, 1.0);
sp6.setSizeHintAlign(-1.0, 0.5);
bx.packEnd(sp6);
sp6.setVisible(true);

sp7 = new elm.Elm.Spinner(win);
sp7.setSizeHintWeight(1.0, 1.0);
sp7.setSizeHintAlign(-1.0, 0.5);
bx.packEnd(sp7);
sp7.setVisible(true);
sp7.setEditable(true);

sp7.on('changed',
  function(obj)
  {
      console.log("Value changed to " + obj.value_get());
  });

sp7.on('delay_changed',
  function(obj)
  {
      console.log("Value delay changed to " + obj.value_get());
  });

win.setVisible(true);
